packages/ilios-common/addon/classes/yup-validations.js

```diff
@@ -0,0 +1,147 @@
+import { tracked } from '@glimmer/tracking';
+import { getProperties } from '@ember/object';
+import { object, setLocale } from 'yup';
+import { restartableTask, timeout } from 'ember-concurrency';
+
+const DEBOUNCE_MS = 100;
+
+/**
+ * Started with: https://mainmatter.com/blog/2021/12/08/validations-in-ember-apps/
+ */
+export default class YupValidations {
+  context;
+  schema;
+  shape;
+
+  @tracked error;
+  @tracked showAllErrors = false;
+  @tracked visibleErrors = [];
+
+  constructor(context, shape) {
+    this.context = context;
+    this.shape = shape;
+    this.schema = object().shape(shape);
+  }
+
+  get errorsByKey() {
+    return this.error?.errors.reduce((acc, validationError) => {
+      const key = validationError.path;
+
+      if (!acc[key]) {
+        acc[key] = [validationError];
+      } else {
+        acc[key].push(validationError);
+      }
+
+      return acc;
+    }, {});
+  }
+
+  get errors() {
+    if (!this.errorsByKey) {
+      return {};
+    }
+    if (this.showAllErrors) {
+      return this.errorsByKey;
+    }
+
+    return getProperties(this.errorsByKey, ...this.visibleErrors);
+  }
+
+  runValidator = restartableTask(async () => {
+    //wait for user input to stop
+    await timeout(DEBOUNCE_MS);
+    const rhett = await this.#validate();
+    //return the result of the validation and not the promise itself
+    return rhett;
+  });
+
+  async #validate() {
+    try {
+      await this.schema.validate(this.#validationProperties(), {
+        abortEarly: false,
```

Comment:
how about setting this to `true`?

`abortEarly: false`

![image](https://github.com/user-attachments/assets/f25e7fc4-09cf-49f5-b9f1-2ad20e090846)



`abortEarly: true`

![image](https://github.com/user-attachments/assets/b5a9232a-5627-4b97-99b0-cf310846faf5)


the latter would be consistent with the error output we're getting from the current implementation.

Reply:
This single field validation case doesn't show it, but what happens is validation dies when *any* field is wrong. So if you had two fields with an error only the first one would show up. I'm more inclined to just take the first error off the stack, but figured we'd see how people feel about seeing two.

Reply:
thanks for the explanation. i now recall you mentioned this before, but it didn't really register with me at the time.